hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/TestStreamWriteOperatorCoordinator.java

```diff
@@ -247,34 +247,16 @@ void testSyncMetadataTable() throws Exception {
     assertThat("One instant need to sync to metadata table", completedTimeline.countInstants(), is(7));
     assertThat(completedTimeline.nthFromLastInstant(1).get().getTimestamp(), is(instant + "001"));
     assertThat(completedTimeline.nthFromLastInstant(1).get().getAction(), is(HoodieTimeline.COMMIT_ACTION));
-    // write another 2 commits
-    for (int i = 7; i < 8; i++) {
+
+    // write another 27 commits to trigger the first clean
+    for (int i = 7; i < 34; i++) {
       instant = mockWriteWithMetadata();
-      metadataTableMetaClient.reloadActiveTimeline();
-      completedTimeline = metadataTableMetaClient.getActiveTimeline().filterCompletedInstants();
-      assertThat("One instant need to sync to metadata table", completedTimeline.countInstants(), is(i + 1));
```

Comment:
I'm skeptical about the cleaning strategy change to MDT, the MDT should have more aggresive/frequent cleaning strategy. Instead of changin the cleaning strategy, can we fix the MDT archival min num commits instead?

Reply:
https://github.com/apache/hudi/pull/8319 modify MDT "hoodie.keep.min.commits“ align to DT's archive config , so I fix it according to alignment logic and just not want to revert the logic of https://github.com/apache/hudi/pull/8319

Reply:
Can you explain again why the default value does not conflict here for MDT with options:
`HoodieMetadataConfig.DEFAULT_METADATA_CLEANER_COMMITS_RETAINED` as 3 and `hoodie.keep.min.commits` as 20.

Reply:
1）Use default data table config, when build MDT's HoodieWriteConfig, HoodieWriteConfig.validate() verify  "hoodie.keep.min.commits"=20 > "hoodie.cleaner.commits.retained"=3, So not conflict
2)   Set data table config: ("hoodie.cleaner.commits.retained", "2"), ("hoodie.keep.min.commits", "3"), MDT's "hoodie.cleaner.commits.retained" set to  HoodieMetadataConfig.DEFAULT_METADATA_CLEANER_COMMITS_RETAINED as 3, then MDT's "hoodie.keep.min.commits"=3  !> "hoodie.cleaner.commits.retained"=3, So conflict

Reply:
Got it, can we set up the MDT `hoodie.cleaner.commits.retained` as:

```java
```

Reply:
That sounds reasonable，I've modified it